src/mod_keystore.erl

```diff
@@ -97,15 +95,14 @@ keys_spec() ->
                                         validate = filename}
         },
         required = [<<"name">>, <<"type">>],
-        process = fun ?MODULE:process_keys/1
+        format_items = map,
+        process = fun ?MODULE:process_key/1
     }.
 
-process_keys(KVs) ->
-    {[[{name, Name}], [{type, Type}]], PathOpts} = proplists:split(KVs, [name, type]),
-    process_key_opts(Name, Type, PathOpts).
-
-process_key_opts(Name, ram, []) -> {Name, ram};
-process_key_opts(Name, file, [{path, Path}]) -> {Name, {file, Path}}.
+process_key(#{name := Name, type := file, path := Path}) ->
+    {Name, {file, Path}};
+process_key(#{name := Name, type := ram}) ->
+    {Name, ram}.
```

Comment:
What if we make them KV items but with the value being a map:
```suggestion
process_key(M) ->
    maps:take(name, M).
```
Of course the code would need to be adjusted a bit.

Reply:
So the key would be the Name, and Value `#{type => ram}` for example?
Maybe I'm misunderstanding, but if so, then I'd rather just skip the process step altogether, and handle maps with `name`, `type` and maybe `path` as keys in the module code, which is certainly a possibility, and could be more elegant 🤔

Reply:
It can stay as it is, the keys make it easy to validate their uniqueness, so I wouldn't treat whole maps as keys.